diff --git a/core/modules/ckeditor/src/CKEditorPluginManager.php b/core/modules/ckeditor/src/CKEditorPluginManager.php
index 735cfa6678..a5212f87ab 100644
--- a/core/modules/ckeditor/src/CKEditorPluginManager.php
+++ b/core/modules/ckeditor/src/CKEditorPluginManager.php
@@ -40,6 +40,24 @@ public function __construct(\Traversable $namespaces, CacheBackendInterface $cac
     $this->setCacheBackend($cache_backend, 'ckeditor_plugins');
   }
 
+  /**
+   * Gets sorted plugin definitions.
+   *
+   * @param array[]|null $definitions
+   *   (optional) The plugin definitions to sort. If omitted, all plugin
+   *   definitions are used.
+   *
+   * @return array[]
+   *   An array of plugin definitions, sorted by label.
+   */
+  public function getSortedDefinitions(array $definitions = NULL) {
+    $definitions = isset($definitions) ? $definitions : $this->getDefinitions();
+    // Sort definitions by key.
+    ksort($definitions);
+
+    return $definitions;
+  }
+
   /**
    * Retrieves enabled plugins' files, keyed by plugin ID.
    *
@@ -66,7 +84,7 @@ public function __construct(\Traversable $namespaces, CacheBackendInterface $cac
    *   For internal plugins, the value is NULL.
    */
   public function getEnabledPluginFiles(Editor $editor, $include_internal_plugins = FALSE) {
-    $plugins = array_keys($this->getDefinitions());
+    $plugins = array_keys($this->getSortedDefinitions());
     $toolbar_buttons = $this->getEnabledButtons($editor);
     $enabled_plugins = [];
     $additional_plugins = [];
@@ -138,7 +156,7 @@ public static function getEnabledButtons(Editor $editor) {
    * @see \Drupal\ckeditor\CKEditorPluginButtonsInterface::getButtons()
    */
   public function getButtons() {
-    $plugins = array_keys($this->getDefinitions());
+    $plugins = array_keys($this->getSortedDefinitions());
     $buttons_plugins = [];
 
     foreach ($plugins as $plugin_id) {
@@ -162,7 +180,7 @@ public function getButtons() {
    *   A configured text editor object.
    */
   public function injectPluginSettingsForm(array &$form, FormStateInterface $form_state, Editor $editor) {
-    $definitions = $this->getDefinitions();
+    $definitions = $this->getSortedDefinitions();
 
     foreach (array_keys($definitions) as $plugin_id) {
       $plugin = $this->createInstance($plugin_id);
diff --git a/core/modules/ckeditor/src/Plugin/Editor/CKEditor.php b/core/modules/ckeditor/src/Plugin/Editor/CKEditor.php
index 49ab1b30dd..5e86fc2328 100644
--- a/core/modules/ckeditor/src/Plugin/Editor/CKEditor.php
+++ b/core/modules/ckeditor/src/Plugin/Editor/CKEditor.php
@@ -185,7 +185,7 @@ public function settingsForm(array $form, FormStateInterface $form_state, Editor
     // necessary for certain filters' (for instance, the html_filter filter)
     // settings to be updated accordingly.
     // Get a list of all external plugins and their corresponding files.
-    $plugins = array_keys($this->ckeditorPluginManager->getDefinitions());
+    $plugins = array_keys($this->ckeditorPluginManager->getSortedDefinitions());
     $all_external_plugins = [];
     foreach ($plugins as $plugin_id) {
       $plugin = $this->ckeditorPluginManager->createInstance($plugin_id);
